### ISSUE TITLE
Strange behaviour with rescuezilla-2.4.1-64bit.jammy.iso

### ISSUE THREAD

**Issue by LaurentGrenet** [ventoy/Ventoy#2139]
### Official FAQ

- [X] I have checked the official FAQ.

### Ventoy Version

1.0.86

### What about latest release

Yes. I have tried the latest release, but the bug still exist.

### Try alternative boot mode

Yes. I have tried them, but the bug still exist.

### BIOS Mode

UEFI Mode

### Partition Style

MBR

### Disk Capacity

4GB (simple USB key)

### Disk Manufacturer

unknown

### Image file checksum (if applicable)

Yes.

### Image file download link (if applicable)

https://github.com/rescuezilla/rescuezilla/releases/download/2.4.1/rescuezilla-2.4.1-64bit.jammy.iso

### What happened?

If I copy the iso file directly on my Ventoy USB key, no problem : boot on rescuezilla iso works fine.
BUT if I have only a vlnk to a local disk on my USB key, and/or if I browse local disks to find the iso, boot on it fails with the message : 
  No bootfile found for UEFI!
  Maybe the image does not support X64 UEFI

While nevertheless it is exactly the same iso file (I have checked the SHA256 of both) that is on my USB key (and works) or on the local disk (and doesn't work).

I don't know if this may be due to a bug, or if there could be a good reason for such strange behaviour, but I really do not understand how it can be possible ????

**Comment by steve6375** [comment]
Maybe faulty flash memory - try another USB flash drive.

**Comment by ventoy** [comment]
Other ISO files have the same behaviour?

**Comment by LaurentGrenet** [comment]
Since it is possible to boot on the ISO, provided it is copied on the USB key, I would have been surprised that problem could come from the key itself, nevertheless I did the suggested test with another USB key : same issue.

And (up to now...) it's the ONLY ISO that generates this problem. In particular, NO PROBLEM with a previous version of Rescuezilla, rescuezilla-2.1.3-64bit.groovy.iso

**Comment by steve6375** [comment]
Just to be clear. Are you saying that if the ISO is on the Ventoy USB drive it works but if the ISO is on a **local disk** it fails?
So the issue is that the ISO fails when running it from a local disk?
If so, what is this local disk? Make/Model/format/compressed filesystem, etc.? 
What if you place the ISO on partition 3 of the USB drive or on a different local disk?

**Comment by steve6375** [comment]
I removed fastboot quiet and nosplash options and booted from ptn3 of Ventoy drive
![image](https://user-images.githubusercontent.com/1469350/211141069-e72b9052-b479-4f15-9952-be5991ebd6de.png)
It hangs here.
boots OK from agFM/grubfm on ptn3
boot OK from Ptn1 (NTFS) of Ventoy drive.

**Comment by steve6375** [comment]
seems OK in latest release 1.0.87  - can you try?

**Comment by LaurentGrenet** [comment]
Hi,

let me clarify some points, try to answer your questions... and share the result of my tests.

1) Clarification : what is exactly the Issue
I boot on my USB key Ventoy.
From the first screen, I can either select an ISO file stored on the USB key itself, or select a vlnk.iso that points to a local drive, or using F2 directly browse local drives to select an ISO
This should provide the same result (allow me to boot on the corresponding ISO), what is the case for all ISO I may have tested recently EXCEPT for one, the rescuezilla-2.4.1-64bit.jammy.iso downloaded from RescueZilla web site.

With this ISO, 
- if it is present on the USB key, boot is OK
- if it is on the local disk, boot fails, whatever it has been selected thanks to the vlnk.iso link, or thanks to direct browsing.

2) What is the "local disk" : It is my standard HD (not SSD) dedicated to backups. Connected in USB. Formatted in NTFS. Not compressed. Works fine AFAIK

3) You suggest me to test with Ventoy 1.0.87 : Just did it : NO CHANGE.

4) Your question about the "local disk" made me wondering : and if there were something wrong on the disk ?
So I did the test to copy the original ISO from the disk where it is to several other disks
- on my "internal" disk (a SSD, NTFS formatted)
- on another "external" SSD, formated in exFAT, connected in USB
- on a 3rd "external" HD (not SSD), formated in NTFS, not compressed, connected in USB

Result : ALL THESE COPIES WERE BOOTABLE !

5) The following test I imagined was then to 
- 5.1 : Copy the "original" ISO to another place ON THE SAME DISK (copy, and not move)
- 5.2 : and also move the original ISO in another place on the same disk (move, this time, and not copy)

Result : The copied file (5.1) is bootable, while the moved one (5.2) remains NOT bootable

6) To be sure that the copies where identical, I calculated the checksum on both, using Nirsoft/HasMyFiles.exe
All hashes were the same, but launch of HashMyFiles ask me a "security question".. because the file was originally "downloaded from Internet"
This makes me to think that this is managed in NTFS with an "Alternate stream data" named "Zone.Identifier"
And if this ASD could be the root cause  (or at least ONE cause) ?

Obviously, if the file is copied to the USB key, or to the disk formatted in exFAT, the ASD is NOT copied. This could explain that ISO copied on the Ventoy USB key is bootable.
But, on the other hand, if the file is copied on a NTFS disk (the same, or another one), all streams are suposed to be copied as well (and they do are).

Nevertheless, I did the test to REMOVE this ASD (in file property, at the bottom, check the "Unblock" box)

Result : the ISO file became BOOTABLE in Ventoy !

7) This do not answer to a question : Why, in such a case, copies of this file, made on the same NTFS disk, that also contain this ASD, are bootable even without "unblocking" them ?
Why the original, and not the copy ? Were there something wrong but invisible in the original file ?

8) To try to have an answer to this question, I downloaded again the iso file from RescueZilla. Like the previous time, I downloaded directly on the same disk.
And I got exactly the same behaviour : the file downloaded is NOT bootable, remains not bootable even if moved (on the same disk...), but its copies (on the same disk) ARE bootable, and if I remove ASD, downloaded ISO file becomes bootable !

Please note that this time, I removed the ASD NOT by checking the "unblock" box in properties, but thanks to (excellent !) Nirsoft/AlternateStreamView.exe, that allowed me to keep a copy of the ASD before removing it. I attach it to this post.
And to also verify that the ASD of the downloaded file is the same than the ASD of its copies...


TO SUM-UP :
Clearly, I have not understood, but clearly also, the issue is more or less linked to the fact that the ISO file comes from a download.
There is a link with the ASD "Zone.Identifier", since if I delete it, issue disappear. But not only, since problem disappear also with copies of the downloaded file that nevertheless have exactly the same ASD....
While problem remain if we only "move" the downloaded file to another place on the same disk.....

We could imagine that "somewhere else" Win10 (and in particular Windows Defender) keeps memory that the ISO file comes from internet and "blocks" something, but since the issue appears when we are booting on the Ventoy key, it appears at a time where neither Win10 nor Defender are active....

And why this REPRODUCIBLE issue with THIS iso, and NOT with other ones, and in particular NOT with rescuezilla-2.1.3-64bit.groovy.iso ....

The only "good news" of these test being that it is "enough" to delete the ASD (either "unblocking" the file, or using Nirsoft/AlternateStreamView.exe to make it bootable with Ventoy...


[rescuezilla-2.4.1-64bit.jammy.iso_Zone.Identifier.txt](https://github.com/ventoy/Ventoy/files/10366757/rescuezilla-2.4.1-64bit.jammy.iso_Zone.Identifier.txt)

**Comment by steve6375** [comment]
Very interesting!
Did you redownload the old groovy iso today or use a previously downloaded file?

**Comment by steve6375** [comment]
Could it be to do with file fragmentation? Downloading directly to NTFS can result in a highly fragmented file , I think because of multiple streams . Moving the file on same disk will not change the file. Copying the file will change fragmentation 
The windows app defraggler will show you fragmentation and allow you to defrag it.

**Comment by steve6375** [comment]
P s. Checksum the file using ventoy ctrl+m,
Maybe grub2 NTFS driver has an issue with the file?

**Comment by LaurentGrenet** [comment]
For "groovy", it's the one I downloaded a long time ago (feb 2021)
About "file fragmentation", I don't think so, otherwise removing alternate stream would take "at least a little time", if it would also "defragment", while it is immediate.

Last (but not least), your hypothesis of NTFS driver having an issue is really interesting, since.... checksum (from inside Ventoy) of the file is 
- impossible in all cases where the file is "not bootable" (Checksum stops at +/- 30% with "error : run list overflow")
- succesful in case of "bootable" file, either bootable because copy of "downloaded" file, or after removing of alternate stream

If Ventoy is not able to "checksum" the file, it is not surprising it is not able to boot on it...

**Comment by steve6375** [comment]
I think it's to do with the $MFT.
It contains attribute records for the file fragments and the web archive attribute. The grub2 NTFS driver might not like the order the attributes are in or it may have an extended $MFT record.

**Comment by steve6375** [comment]
So is the file highly fragmented? Check using defraggler?

**Comment by LaurentGrenet** [comment]
I guess (even if I'm not sure) that $MFT contains also directly the "alternate streams", and this could explain that removing one ASD forces rewriting of corresponding $MFT entry, making it more "compatible" with NTFS driver expectations.

Nevertheless, it remains a bug for Ventoy, since these files are OK from Windows standpoint (checksumming under Windows works fine, it's only checksumming under Ventoy that fails...), and it would be great that a solution is found in Ventoy... even if it's not a "blocking bug" since removing of ASD removes the problem.

**Comment by steve6375** [comment]
NTFS grub2 driver is known to have bugs.
Defragging would probably permanently fix the file
To fix, we need to know all bytes in the MFT record for that file as it is obviously different from other similar files.

**Comment by LaurentGrenet** [comment]
I do not have defraggler, so I didn't check it.
But as I said before, I cannot imagine one second that fragmentation may be the cause, since otherwise "only" remove the ASD (that surely DO NOT defrag the main "stream" : it doesn't take enough time to do it) woudn't remove the issue.
Moreover, according to "windows native" defragmentation tool, the corresponding disk is not fragmented at all.

**Comment by steve6375** [comment]
Removing the asd would merely remove the attribute record in the file MFT record. It just rewrites a few sectors.
The grub2 driver may then be able to parse the file record without error as the order of attributes or something else will change.
If the file is fragmented or have multiple attribute entries and extends the 1024 byte MFT file record then it might confuse the NTFS driver. Knowing if the file is fragmented would help our understanding.
Without investigation of the bad file MFT record we will not be able to get a better idea of where the issue may lie.

**Comment by LaurentGrenet** [comment]
To investigate, the best would be you reproduce the issue on your own laptop. I'm afraid that it would be very difficult to investigate MFT record on MY laptop...

BTW, did you try to reproduce the issue ? And do you have the same ?

**Comment by steve6375** [comment]
Can you reproduce the issue by downloading a 2nd copy ? I suspect you will at least need to download onto the same NTFS disk directly using your browser in same way as you did before.
I suspect, since no other similar iso files have this issue, you will find it hard to reproduce.

**Comment by LaurentGrenet** [comment]
During my tests, I have downloaded at least 5 or 6 times this same file. Always to the same NTFS HD (not SSD), and always the same way : clicking on the link in Rescuezilla website under Firefox, and selecting "save as..." 
Saving it in a different place at each test.
And the 5 or 6 times, I've got the same issue.

On the other hand, I did once the same test except I downloaded the file on my internal disk, that is also NTFS, but is a SSD (and NOT a HD).
Only in this case, the downloaded file was "directly" bootable, neither requiring to "copy" it, nor removing its "alternate stream".

**Comment by steve6375** [comment]
Oh that is interesting! No, I can't reproduce even with direct download.
What exact page did you download it from?
https://rescuezilla.com/download ?
Presumably you were running FireFox under Windows 10?
Do you get same issue if you just click on the link to download?
Same issue if you use Chrome or Edge to download?
What is the size of the NTFS HD? Have the partitions on it been resized at all in its history (after any wiping and partitioning)? How was it formatted? Anything atypical about it or its history?

**Comment by LaurentGrenet** [comment]
1) In the page https://rescuezilla.com/download, I click on the link that is in the 1st paragraph, and whose URL is https://github.com/rescuezilla/rescuezilla/releases/download/2.4.1/rescuezilla-2.4.1-64bit.jammy.iso

2) Yes, this is done with Firefox 108.0.2, under W10 22H2, build 19045.2364 (ie. last update)

3) I do not understand your question : "Do you get same issue if you just click on the link to download?" : Actually, I just click on the link, but Firefox opens the popup asking me if I want to open it or to store it, and allowing me to choose where to store it...
This because FFx parameters is "Always ask you where to save files"

4) I just did a test downloading it with Edge... and the file downloaded **is directly "bootable" !** No need to copy it or to remove ASD...

5) NTFS HD : SILICON POWER DIAMOND D05 1000GB USB 3.0
Size = officially 1Tb, actually 931 Gb reported by Windows, ie. 1000202235904 bytes - Remaining free space  +/- 140 Gb
Only one partition
Formatted once when I bought it a long time ago (Oct 2012), or maybe it was already formatted, I do not remember.
Never reformatted nor repartitioned since
No defect nor warning according to CrystalDiskInfo

**Comment by steve6375** [comment]
OK, Edge not giving a problem is interesting. I suspect that Chrome will be similar to Edge and also not cause any issues?
I did notice that when downloading using FireFox, it temporarily makes two files in the target folder. One file has the same name but the other file seems to be a temporary file of some kind. In my case the resultant file had 8 fragments (as reported by Defraggler which is easily installed onto Win10 ;-).

**Comment by LaurentGrenet** [comment]
I do not have Chrome (and prefer do not install it...), so I cannot do the test with Chrome

About files produced by Firefox downloads, AFAIK when Ffx downloads a file, it creates an empty entry with the final name (to avoid another file with this name can be created during download), downloads the file under a temporary name, and at the end, removes the empty entry and rename the downloaded file with its final name. 

And I think that Edge does the same, except it doesn't create temporary empty entry with the final name. But download the file under a temporary name, and at the end, renames it with the final name.

**Comment by steve6375** [comment]
Since you have a 'good' jammy iso (saved using Edge) and a 'bad' jammy iso (saved using FireFox) you can use WinHex to examine the two file MFT records and try to spot any differences.

1. Install WinHex
2. Run as Admin
3. Options - Edit mode - Read Only
5. Press F9 (Tools - Open Disk)
6. navigate to drive letter and the iso file you want to examine and select it in top pane
7. right-click - Navigation - Go to File Record
8. You should now see MFT record which begins with FILE0
![image](https://user-images.githubusercontent.com/1469350/211188560-2c2c6fea-29b2-4f2d-bff6-38107efef983.png)
each record is 2 sectors (1024 bytes) long, so next file record will start with FILE0.
So see if the 1024 bytes for the jammy iso is completely full of data - if so it may be an extended record and you will need to find the next record (not sure how to do that yet??). Hopefully the MFT record will not be larger than 1024 bytes to make it easier!
maybe you could post screenshow of the 1024 bytes starting with FILE0 for the two iso files (good and bad)?
If you have more than one 'bad' jammy ISO, maybe you could find the one with the smallest MFT record data to make it easier to try and spot what grub2 is having a problem with?
e.g. in my file (screenshot above shows most of it), the 1024 byte (2 sector) record contains approx  1.5 sectors worth of data.

**Comment by LaurentGrenet** [comment]
> Since you have a 'good' jammy iso (saved using Edge) and a 'bad' jammy iso (saved using FireFox)

Actually, I **used to have**.... but I have no more... I purged both of them...

OK. I redownloaded them once more.
What is new this time, is that **BOTH of them**, are now "non bootable", **inc. the one downloaded with Edge**.

Nevertheless, as we know that it is "enough" to either copy a non-bootable iso to another place to make this copy bootable, or to remove ASD in a non-bootable iso to make it bootable, I think I can do the investigation you are asking for.

But before doing it, I have a question : What is exactly the WinHex you want I install ? Is it the one we can find here : https://www.x-ways.net/winhex/index-m.html
I ask the question, because according to this website, it seems that a paying licence is required to use it... Maybe it is usable "for free" with limited features ?
So before trying to download it and to install it, I'd like your confirmation.

In the mean time, I keep carefully unchanged / untouched the two "non-bootable" copies I've got (in the total, I guess I have downloaded this iso more that 10 or 15 times in the past 3 days.... 17' per download... More that 3 or 4 hours downloading...)

**Comment by steve6375** [comment]
I used the free evaluation version as you can tell from the screenshot

**Comment by LaurentGrenet** [comment]
OK, pls find here results of this 1st investigation.

Let me remind the situation : I have two "not-bootable" files, one downloaded with Ffox, the second downloaded with Edge.
I have also a copy done after download of each of these files. These copies, done in the Windows explorer, are "bootable"
All these files have currently an "Alternate Stream Data" named "Zone.Identifier", since result of a download.
We know that if we remove this supplementary ASD from a "non-bootable" file, it makes it "bootable". But since this is a non-reversible action (I cannot "reinject" such ASD to make the file non bootable again), I have not done it on any of these files. Maybe it will be interesting to do it later, but I won't do it before you ask me to do it.

For each of these 4 files (2 non-bootable, 2 bootable), with WinHex in Read-Only Edit mode, I navigate to the File Record, and each time I took a copy of the 1024 bytes starting with FILE0 : copy to a new file
These 4 copies are uploaded hereafter. Names are self-explanatory...
I'm now waiting for supplementary tests to do that could help you....

[WinHex extracts.zip](https://github.com/ventoy/Ventoy/files/10369323/WinHex.extracts.zip)

**Comment by LaurentGrenet** [comment]
Hi @steve6375 
no answer to my post, does it mean :
- that you haven't seen it ?
- or that you haven't time enough yet to analyze it ?
- or that it's now OK for you ?

In other words, **do I have to keep the files, or may I drop them** ?

**Comment by steve6375** [comment]
sorry, yes please keep them. I have been a bit busy. Will try to take a closer look in next few days.

**Comment by LaurentGrenet** [comment]
OK, no problem. Now I know you are still on the topic, and I'm not in a hurry ! Take your time.

**Comment by steve6375** [comment]
I have had a look and the bad File records seem to contain an extra attribute called $ATTRIBUTE_LIST  0x20.
https://flatcap.github.io/linux-ntfs/ntfs/attributes/attribute_list.html
![image](https://user-images.githubusercontent.com/1469350/212183652-ebe2b742-605b-4f4c-a55d-bce8cb730222.png)

The desription says this is very rare and so I am guessing that this is not handled correctly by the grub2 NTFS driver.
I am not sure how or where to report this to the grub community but there should be enough to go on here.

**Comment by steve6375** [comment]
If you legacy boot to Ventoy and use ctrl+m - does the 'bad' file also give an error? If so that would possibly mean both the EFI NTFS driver and the legacy NTFS driver have the similar issues. If not then it means it is a EFI only bug.
Do you know if the UEFI BIOS on the test system contains an NTFS driver as well as the default FAT driver? Some UEFI BIOSes in Asus systems for instance, can natively UEFI boot from an NTFS partition as the UEFI BIOS already includes an NTFS driver.
Also, the problem may possibly be in your UEFI BIOS.
Is there any way you can connect that disk which has the 'bad' file to a different system (or boot Ventoy from a Virtual Machine and attach the disk to the VM) and then do the ctrl+m test?

**Comment by steve6375** [comment]
P.S. Can you UEFI-boot from say a mainstream Ubuntu ISO just to the Ubuntu grub2 menu, then press c and then check the 'bad' jammy iso file perhaps using something like the grub2 cmp command
e.g.
`cmp (hdN,Y)/FFjammy.iso (hdN,Y)/FFjammy.iso `

I assume this would fail under Ventoy grub2 (?can you check?) so it would be interesting to know if it fails under the grub2 used by Ubuntu.

**Comment by steve6375** [comment]
Under Ubuntu you could try the ntfsinfo command on the bad file, e.g.
`ntfsinfo -v -F osimg/FFbadjammy.iso /dev/sdc3
`
and add the output into your post here so we can see the nicely parsed information too?

I have found one previous grub bug report
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=943544  which seems to trace the bug to a line of code in the ntfs.c module
```
it fails here in ntfs.c:

      if (ofs + len > u32at (pa, 0x10))
        return grub_error (GRUB_ERR_BAD_FS, "read out of range");
```
and some rather mind-boggling details on the $ATTRIBUTE_LIST
https://stackoverflow.com/questions/42777907/understanding-the-attribute-list-in-ntfs

**Comment by steve6375** [comment]
I also just also noticed that both 'bad' files also have an extra 0x40 attribute record.
https://flatcap.github.io/linux-ntfs/ntfs/attributes/object_id.html
for which there are two definitions for the same attribute...

<html>
<body>


0x40 | NT | $VOLUME_VERSION
-- | -- | --
0x40 | 2K | $OBJECT_ID


</body>
</html>


Strangely the notes say
https://flatcap.github.io/linux-ntfs/ntfs/attributes/index.html

$VOLUME_VERSION existed in NTFS v1.2, but weren't used. They no longer exist in NTFS v3.0 (that used by Win2K).
but $OBJECT_ID is supposed to be 0x40 bytes long ??? but it is actually 0x20 bytes in the bad file record ???
This might be tripping up grub2 ntfs driver ??

**Comment by LaurentGrenet** [comment]
Hi @steve6375 
first answers to your questions : 

1) Yes, as already mentioned, with all ISO that are not bootable, checksumming run under Ventoy (ctrl+m) **systematically fails**, with the following error message : 
              _Calculating MD5 xx%  error: run list overflown.
              VVVVVVVV 1 grub_errno=0_

On the other hand, with all ISO files that are bootable, checksumming run under Ventoy (ctrl+m) is OK

Please note that checksumming run directly under Win10 (using Nirsoft HasMyFiles.exe) is OK for ALL files, **the "bad" as well** as the "good".

Please note also that the checksumming error of "bad" files occurs either when booting with Ventoy my "physical/real" laptop, or when booting (with Ventoy) a Virtualbox VM.

Just FYI, I'm able to boot succesfully "bad" iso files on a Virtualbox VM if I'm NOT using Ventoy, **but loading directly the "bad" ISO file in the virtual CD drive**. (and also, of course, "good" ISO files)  

2) I'm not able to answer your question "_Do you know if the UEFI BIOS on the test system contains an NTFS driver as well as the default FAT driver?_" 
No idea about that. I can only say you that the "physical" test system is a Dell Precision 3520
You may possibly find the answer to your question on Dell website.

3) Next questions around Ubuntu : I do not have (yet) Ubuntu ISO. So I'm right now downloading the last LTS ISO file, but it's very very long to download it (still more than 5 hours to wait...) . When I'll got it, I'll do the test you are asking for.

4) About the possible cause due to the extra attribute ($ATTRIBUTE_LIST 0x20), do you have **verified** that this extra attribute  is present **ONLY** in the "bad" files (and present **in the TWO** "bad" files), and NOT in any of the "good" ones, that are, let me remind you, copies of the "bad" ones done in Windows Explorer ?
5) Moreover, let me remind you that we saw in previous tests that removing the "Alternate Stream Data" in a "non-bootable" file makes it to become bootable. Since we have two "non-bootable", do you want I "sacrifice" one of them removing its ASD, to check if it makes to disappear the extra attribute from its $MFT ?

**Comment by steve6375** [comment]
Can you clarify about Q1 please? If you LEGACY BOOT to Ventoy (not UEFI boot) does ctrl+m still fail on the bad file? i.e. both UEFI and legacy grub2 fail to read the bad ISO.

**Comment by steve6375** [comment]
re. Q4, the 0x20 and 0x40 attributes are only in the two 'bad' iso files.
re. Q5 - we might as well remove the alternate Stream Data from one of them and see what we get. I suspect that it may be the same as the copied files but there is only one way to find out and that is to try it!

**Comment by LaurentGrenet** [comment]
> Can you clarify about Q1 please? If you LEGACY BOOT to Ventoy (not UEFI boot) does ctrl+m still fail on the bad file? i.e. both UEFI and legacy grub2 fail to read the bad ISO.

Hi,
I surely never say that.... since I was not even aware that it is possible to boot my laptop in legacy mode. All my tests up to now are done in UEFI boot mode.
Due to your clarification request, I tried to boot in Legacy mode... but I didn't succeed...
In the boot menu I may display at boot, under the option "Legacy boot", the USB key is not proposed. Only a more "generic" "USB Storage device", ad a "Onboard NIC".
If I select the 1st one, the system doesn't boot but rather gives an error : no bootable device found...
And if I select the 2nd one, the system says it is waiting for something... that never ends.

There is also under "Other Options" : "Change boot mode". I tried to activate it, and select Legacy boot as default, but the system asks for so many confirmations, the first in yellow, the last in red, stating "be careful, are you sure, etc..." with remarks/explanations I didn't really understand, that it frightened me, and I didn't go to the end, and didn't activate the Legacy boot mode as default....

I tried also to "legacy boot" a Virtualbox VM : I unchecked on its parameters the use of EFI, but at boot, it doesn't propose me the USB key to boot, but only hard disk and CD. 

So, if it's really feasible WTHOUT ANY RISK pls explain me in a detailed way how I can "LEGACY BOOT to Ventoy".

**Comment by LaurentGrenet** [comment]
About the tests with Ubuntu : 

Pls be a little bit more specific (no, not a little bit more, but VERY specific) about command to pass during Ubuntu boot, and when Ubuntu is running.
I'm not experienced at all in this domain, and I'm not able to "translate" your commands to my specific environment.
In particular, I guess that when you write : 
`cmp (hdN,Y)/FFjammy.iso (hdN,Y)/FFjammy.iso `
FFjammy.iso is supposed to be the name of the "bad" iso file, but how to tell Ubuntu where this file is stored ???
Moreover, I guess that cmp is for compare... and both args are identical... sounds strange....

The same for the ntfsinfo command to pass, Ubuntu running.

Thank you for detailed explanations if you want I can help you....

**Comment by steve6375** [comment]
Here is an example of the cmp command in the ventoy grub2 shell (press c in ventoy menu)
![image](https://user-images.githubusercontent.com/1469350/212369873-618fe1e7-24e1-4322-8738-d582c4ed3004.png)

I use search.file to find which drive and partition the iso is on
Then I use the cmp command to compare it with itself
If there is a problem, then it should fail.
So first try it in Ventoy grub2 - we expect it to fail on a 'bad' jammy iso, but pass on a good copy.

Now, use Ventoy to select the Ubuntu ISO - use Normal option
Next you will see the grub2 boot menu for the Ubuntu ISO
Do not boot to Ubuntu - simply press c to get to the grub2 shell which is used by the ISO.
Then repeat the cmp commands.
If you get an error then it means the NTFS driver bug is also in the version of grub2 used to boot the Ubuntu ISO.
If you get an error in Ventoy grub2 but not in Ubuntu grub2 then it means the NTFS UEFI64 grub2 driver use by Ventoy has a bug.

**Comment by steve6375** [comment]
Boot to an Ubuntu ISO
Choose Try Ubuntu
press SHIFT+ALT+t  for terminal
type sudo fdisk -l    to list disks
find the one which has the bad jammy iso on it (use file explorer on desktop if required)
unmount the partition using  sudo umount /dev/XXXXX
run the ntfsinfo command similar to one below on bad jammy iso
![image](https://user-images.githubusercontent.com/1469350/212373123-256f7084-d93a-4fa9-8bf3-97409d5b1622.png)

**Comment by LaurentGrenet** [comment]
> Maybe you could try legacy and uefi booting on a different computer or a friend's?
> It does sound like it is a problem with the ventoy grub2 NTFS uefi64 driver so you should get same issue on any system. If so, the next thing might be to see if same issue is in legacy NTFS driver.

I did the test on another computer, W7, **NOT UEFI**
Since it is a quite old laptop, I was afraid it was not able to boot on USB. But actually, it is able to do so.
And the result is that checksumming of "bad" ISO is NOT succesful (with same error than on UEFI / W10 laptop), while checksumming of "good" ISO is OK.
But the main difference with my own laptop is that on this old one, "good" ISO are NOT BOOTABLE (under Ventoy) : Error message is as follow 
```
error: File (hd0,1)/---------------VENTOYVLNK.vlnk.iso is not bootable.
VVVVVVVV 5 grub_errno=0
chain empty failed
press Enter to continue .......
```

On the other hand, trying to boot on a "bad" file generates almost the same error message, except it begins with 3 supplementary line 
```
error: Unsupported chunk list.
.
VVVVVVVV 0 grub_errno=0
error: File (hd0,1)/---------------VENTOYVLNK.vlnk.iso is not bootable.
VVVVVVVV 1 grub_errno=0
chain empty failed
press Enter to continue .......
```

I'll do the tests with Ubuntu in a couple of hours, and I'll keep you informed.

But I can already inform you that **I have exactly the same problem with Ubuntu ISO that I have with Rescuezilla ISOs** : 
- The file directly downloaded is NOT bootable, (nor "Checksum"-able)
- but its copies created under Windows Explorer ARE bootable (and "checksum"-able)
- and if we only remove its ASD, it becomes bootable (and "checksum"-able)

All these considerations are on my Laptop, W10/UEFI

**Comment by KaMyKaSii** [comment]
I believe I have the same problem. I have a Linux Mint 21.1 iso on my Ventoy external HD with NTFS that boots to GRUB but not the system, the "out of range" error occurs. When trying to checksum through Ventoy, the same error occurs. On a booted system, the file checksum proceeds correctly and returns the expected value for a healthy file. Using that same file as a source for Etcher to write it to another USB stick it is possible to make Linux Mint boot correctly, so I think the assumption that Ventoy does not support booting fragmented files from NTFS partitions makes sense. I can even boot into Windows and try to defrag the partition to see if that solves the problem but it's a completely full 1TB partition and my computer only has USB 2.0 ports, I'm afraid it will take forever.

<details>
  <summary>Screenshots</summary>

![IMG_20230113_120117](https://user-images.githubusercontent.com/17092335/212382389-a19f7bdc-1521-41a7-9513-03432cbf329f.jpg)
![IMG_20230113_123908](https://user-images.githubusercontent.com/17092335/212382410-f2c379bb-c554-4bd4-af48-a9815876da1b.jpg)
![Screenshot from 2023-01-13 14-34-17](https://user-images.githubusercontent.com/17092335/212382992-23889f51-123b-496e-a395-36ee70def3d3.png)
![Screenshot from 2023-01-13 14-34-51](https://user-images.githubusercontent.com/17092335/212383037-2f485c90-7be8-440e-8744-54dcdc126fea.png)

</details>

**Comment by steve6375** [comment]
It should only happen if you downloaded the file directly to the NTFS usb drive partition.
If you make a copy of the file then you should not get any problem.
Are you legacy booting or uefi booting

**Comment by LaurentGrenet** [comment]
Hi @steve6375 
hereafter the result of my tests "around ubuntu"

0) All these tests have been run on a VirtualBox VM (and not on my physical laptop)

1) In Ventoy Grub, the cmp command never ends (and in particular never explicitely failed with an error message) when applied to "bad" iso. I need to reboot once again to recuperate the hand...
On the other hand, applied to "good.iso", it answers quite quickly "files are identical."
I did the test also on "Good", because I didn't know if I had to wait longer when applied to "Bad".

2) Unable to run the same on Ubuntu Grub, since it didn't have access to my USB drive in this environment. In particular, the search.file didn't find the ISOs. Maybe this is due to the fact that the ubuntu.iso on which I was booting is on the same disk ?

3) I then boot again on Ubuntu, and selected "Try Ubuntu"
From Ubuntu running, I've been able to run the ntfsinfo command on both bad and good iso.
I attach to this post the two results.

[Bad.ntfsinfo.txt](https://github.com/ventoy/Ventoy/files/10415347/Bad.ntfsinfo.txt)
[Good.ntfsinfo.txt](https://github.com/ventoy/Ventoy/files/10415348/Good.ntfsinfo.txt)

Pls tell me if you need more.

**Comment by steve6375** [comment]
2. If the ISo is on the same partition then you need the LINUX_REMOUNT option set in the ventoy.json file.
I had this already set on my USB drive.
Please try again (use ventoyplugson.exe to set this for legacy and uefi).
Here is my file.

```
{
    "control":[
        { "VTOY_WIN11_BYPASS_CHECK": "1" },
        { "VTOY_DEFAULT_KBD_LAYOUT": "QWERTY_UK" }
    ],
    "control_legacy":[
        { "VTOY_WIN11_BYPASS_CHECK": "1" },
        { "VTOY_LINUX_REMOUNT": "1" }
    ],
    "control_uefi":[
        { "VTOY_DEFAULT_MENU_MODE": "1" },
        { "VTOY_WIN11_BYPASS_CHECK": "1" },
        { "VTOY_LINUX_REMOUNT": "1" }
    ]
}

```

**Comment by LaurentGrenet** [comment]
I did what you ask for... but it doesn't work. ISOs (bad and good) on the same partition than the Ubuntu ISO are not "seen".

But actually, I'm not sure that my ventoy.json is taken into account, because I took the opportunity of using VentoyPlugson.exe to also set Default_KBD to French, and Menu_Language to fr_FR... and nevertheless, menus are in English, and keyboard is still qwerty...
VentoyPlugson.exe created the ventoy.json in a ventoy directory, on root of Ventoy USB key, whose content is : 
```
{
    "control":[
        { "VTOY_LINUX_REMOUNT": "1" },
        { "VTOY_DEFAULT_KBD_LAYOUT": "FRENCH" },
        { "VTOY_MENU_LANGUAGE": "fr_FR" }
    ],
    "control_legacy":[
        { "VTOY_LINUX_REMOUNT": "1" }
    ],
    "control_uefi":[
        { "VTOY_LINUX_REMOUNT": "1" }
    ]
}

```

**Comment by steve6375** [comment]
You can use one of the Ventoy f keys to check the JSON file... F7? Can't check at moment

**Comment by LaurentGrenet** [comment]
Actually, it's under F5 menu, but no problem...
I did it, and here is what it displays
![2023-01-14 19_59_42-Dell-Laurent (Tests boot Ventoy)  En fonction  - Oracle VM VirtualBox](https://user-images.githubusercontent.com/32399301/212492324-96d7d5bd-a6c6-4b7f-bb68-870eedb142d4.png)

So this means that neither KBD_layout nor Language were taken into consideration....

I then re-run VentoyPlugson.exe, and saw it was possible also to put these values not only in "global control", but also in "Legacy control" and in "UEFI control". That I did.
Now my ventoy.json file is as follow
```
{
    "control":[
        { "VTOY_LINUX_REMOUNT": "1" },
        { "VTOY_DEFAULT_KBD_LAYOUT": "FRENCH" },
        { "VTOY_MENU_LANGUAGE": "fr_FR" }
    ],
    "control_legacy":[
        { "VTOY_LINUX_REMOUNT": "1" },
        { "VTOY_DEFAULT_KBD_LAYOUT": "FRENCH" },
        { "VTOY_MENU_LANGUAGE": "fr_FR" }
    ],
    "control_uefi":[
        { "VTOY_LINUX_REMOUNT": "1" },
        { "VTOY_DEFAULT_KBD_LAYOUT": "FRENCH" },
        { "VTOY_MENU_LANGUAGE": "fr_FR" }
    ]
}

```
and my menus are in French, and my Keyboard is french layout.
And if I run again the F5 menu, display of json is now as follow.

![2023-01-14 20_09_58-Dell-Laurent (Tests boot Ventoy)  En fonction  - Oracle VM VirtualBox](https://user-images.githubusercontent.com/32399301/212492337-29b74afa-2b64-466c-919b-6f6aa26d42cd.png)


But, **and it is the important point**, when I boot on Ubuntu, then "c" to go to bash under grub, I continue to do not have access to the ISO files....

**Comment by steve6375** [comment]
Sorry, I wasn't really looking hard at you keyboard and language settings. It is very confusing but if you have ANY entry in a different section, e.g.  control_efi, then only those settings are used when you uefi boot. Same for legacy section.
I don't understand why you can't access the Ventoy partition as I had no problem Using vbox+vmub under win10.

**Comment by steve6375** [comment]
I don't think the remount in JSON will affect the Ubuntu grub2 anyway
Are you sure you can't access the partition at all under Ubuntu grub2? Not even using an ls command?

**Comment by LaurentGrenet** [comment]
I just did a new test : I put the Ubuntu ISO on another disk.
Then I boot on it.
But when, from grub2 boot menu for the Ubuntu ISO I press "c" to get to the grub2 shell, from this shell I cannot "see" the ISO files, even if they are not on the same disk than the Ubuntu ISO I'm booting on...

Just one remark : you write "_I don't understand why you can't access the Ventoy partition ..._" : Let me remind you that these files are NOT on the Ventoy USB key, but on a NTFS USB key that is NOT the Ventoy disk...  If they were on the Ventoy USB key, there wouldn't be any problem to boot on them... Go back to the initial description of the problem...

I'm afraid we are not far from the end of tests I may run...

Do you want I run now the ntfsinfo on the "Bad.iso" from which I would only remove the ASD (that makes it to become good...) to compare the ntfsinfo result of the same file with and without ASD ?

Or do you have another test to ask me before ?

**Comment by LaurentGrenet** [comment]
> Are you sure you can't access the partition at all under Ubuntu grub2? Not even using an ls command?

No idea about that. Tell me which command (and syntax, I'm not skilled in Linux) I may use to do that.
The only thing I say, is that when, on your recomendation, I type "search.file /Bad.iso" to get the reference of the disk where it is, I get the answer "not found" (or something like that....)

**Comment by steve6375** [comment]
Did you specify the correct path as well as correct filename for file.search?
ls is standard lower case LS command
E.g.
ls (hd1,1)/

**Comment by LaurentGrenet** [comment]
For file.search, yes, the file is Bad.iso and is on the root of the disk 
And since I do not have answer from search file, I do not have the reference of the disk... and cannot run the ls command.

Is there any command (available in this "limited bash") that could allow me to get the list of available disks. Or maybe to mount one of these disks that wouldn't be actually mounted ?

**Comment by LaurentGrenet** [comment]
With a ls -l, I've been able to see that my disk is (hd1,msdos1)
But if I type
ls -l (hd1,msdos1)/
I can see the different folders present on the disk, but NOT the iso files.....

I browsed several folders in this disk, and it looks like I can "see" files ONLY if they are not "too big"

**Comment by LaurentGrenet** [comment]
Not able to say where is the limit, but clearly when I browse a folder that contains several files, most of them under 350 Mb, but also one of 1.90Gb and one of 4.32 Gb, ls -l gives me all the files EXCEPT the one of 1.9G and the one of 4.3G....

**Comment by LaurentGrenet** [comment]
This explain why I cannot "see" the Bad.iso file that is 1.07Gb...
and then cannot run a cmp of it with itself...

**Comment by LaurentGrenet** [comment]
I repeat my proposal : 

Do you want I run now the ntfsinfo on the "Bad.iso" from which I would only remove the ASD (that makes it to become good...) to compare the ntfsinfo result of the same file with and without ASD ?

Or do you have another test to ask me before ?

**Comment by steve6375** [comment]
This is getting even more weird! I can see all files with ls -l
presumably Ventoy grub2   ls -l  works OK?

Do you get same resuly on a real system in Ubuntu grub2 rather than a VM?

P.S. I am using Ubuntu 21.04 desktop amd64 ISO (grub2 menu) for my tests.

as for ntfsinfo test - it would be better to wait until we can finish these tests because we would only have one 'bad' file which could be accidentally fixed and then we would have none unless you could make some more bad files easily?

**Comment by steve6375** [comment]
can't remember, but did you try the cmp command by specifying (hd1,msdos1)/xxxxx  for both filenames  under Ubuntu grub2 even though search would not work, since you know where it should be?
cmp (hd1,msdos1)/badjammy.iso (hd1,msdos1)/badjammy.iso

**Comment by LaurentGrenet** [comment]
> presumably Ventoy grub2 ls -l works OK?

Yes



> Do you get same result on a real system in Ubuntu grub2 rather than a VM?

Yes

> P.S. I am using Ubuntu 21.04 desktop amd64 ISO (grub2 menu) for my tests.

for me it's ubuntu-22.04.1-desktop-amd64

> can't remember, but did you try the cmp command by specifying (hd1,msdos1)/xxxxx for both filenames under Ubuntu grub2

Don't remember as well, so I just did the test, and the answer is immediate : error: out of memory

> it would be better to wait until we can finish these tests because we would only have one 'bad' file

No, I have TWO bad iso files. Current tests are run on the one downloaded with Edge, but I have a second one, downloaded with Firefox

**Comment by LaurentGrenet** [comment]
I stop here for this evening. I'll read your comments tomorrow.
Have a good night !

**Comment by steve6375** [comment]
Thanks very much for your help. if the Ubuntu grub2 cmp fails then there is nothing more to do except to see how what fixing a bad file actually does to the file record using ntfsinfo. So go ahead and try it on one of the bad files. Have a good evening. :-)

**Comment by steve6375** [comment]
Sorry - does the Ubuntu grub2 cmp test work OK on a different, similarly sized ISO on the same partitition?
i.e. is it failing because of a bad file or just because it cant handle large files in that version of grub2?

**Comment by LaurentGrenet** [comment]
Hi @steve6375 

yes, you're right, I missed to check that the cmp test worked fine, or not, on a "good" iso with a similar size.

So, I just did the test, and the result is the same with "good" or "bad" iso.
But, even more, doing this test, I discovered incidently that the "automatic completion" in Bash **actually showed** the "big iso" files, not shown by the ls command.
And moreover, a ls command with these files as parameter... generate the "out of memory" error (the same than with cmp command).
Here a screen copy that better explain : 
![2023-01-15 09_51_49-Dell-Laurent (Avec ISO ubuntu sur D_)  En fonction  - Oracle VM VirtualBox](https://user-images.githubusercontent.com/32399301/212542940-7d72ec17-89f4-4cca-b1e4-56dfdb0e16a6.png)


Then I ran the ntfsinfo on a "bad but repaired" ISO.
I did these tests on the "bad" ISO I downloaded with FireFox, ie. physically another file than the one used in previous tests, that was issued of a download with Edge.

1) I have verified that the Bad.iso was actually NOT bootable with Ventoy : OK, it is not bootable
2) Then I ran the ntfsinfo on it : Result in attached file "Bad.ntfsinfo.txt"
[Bad.ntfsinfo.txt](https://github.com/ventoy/Ventoy/files/10420287/Bad.ntfsinfo.txt)
3) Then I removed the ASD, using the "Unblock" check box in "Properties" (in W10 Windows explorer) 
 
![2023-01-15 14_22_41-Propriétés de _ Bad iso](https://user-images.githubusercontent.com/32399301/212543300-84ed9d18-40dd-496a-8efd-e8439fdb1d4e.png)

4) Then I verified that removing of ASD was enough and that this file (still named Bad.iso) BECAME ACTUALLY BOOTABLE with Ventoy : OK, boot is now succesful
5) And last, I ran the ntfsinfo on it : Result in attached file "Bad.repaired.ntfsinfo.txt"
[Bad.repaired.ntfsinfo.txt](https://github.com/ventoy/Ventoy/files/10420290/Bad.repaired.ntfsinfo.txt)

I think that we are at the end of tests I may run. Nevertheless, I keep during some time (1 or 2 weeks) all the materials I used for these test (files, and VM), so if you think to a new test, pls ask ! 
But after 1 or 2 weeks, I'll remove all these data... because they use quite a lot of space !

I hope that these tests will help you to find a solution to the problem.

PS : I just have an idea (not even sure it makes sense....) : If I have the problem on a repetitive way, but you are not able on your side to reproduce it, could it be due to the fact that my NTFS disk and yours are not using the same version of NTFS ?
This could also explain that I didn't reproduce the issue when I download not on my USB NTFS disk, but on my internal NTFS disk...
But, since I don't even know if  this concept of NTFS version exists and makes sense, so I'm defnitely not able to check which version of NTFS my two disks (USB and internal) are using.
If you think it makes sense, and if you know how could I check these version levels, don't hesitate to tell me...

**Comment by steve6375** [comment]
Not sure about NTFS version issues. 
I am having problems comparing the ntfsinfo outputs between the bad and badfixed output. I can't decipher the changes that well and of course ntfsinfo may also have bugs.
Sorry to ask, but could you also do a 1024 byte dump of the file record as before but of the badfixed file?
Also, do you have a directly downloaded jammy iso using edge or Firefox that does work with Ventoy? Not copied, just downloaded directly in same way as a bad file.
It would be good to look at a hex dump of that file record too.

**Comment by LaurentGrenet** [comment]
> Also, do you have a directly downloaded jammy iso using edge or Firefox that does work with Ventoy?

No, I do NOT have. None of the files I have downloaded were directly bootable.
I tried to download it once more (with Edge), but this n+1th downloaded file was not bootable.
Nevertheless, I took the opportunity of this new download to send you the HexDump of the **SAME file**, before and after unblocking.

1) Download the file
2) Verify it is NOT bootable
3) Take a copy of 1024bytes HexDump of $MFT : HexDumpBeforeUnblock
4) "Unblock" the file (ie. remove the ASD)
5) Verify that the file ACTUALLY BECAME bootable
6) Take a new copy of the 1024bytes HexDump of $MFT : HexDumpAFTERUnblock
7) Send you these HexDump 
[HexDump before and after unblock.zip](https://github.com/ventoy/Ventoy/files/10421081/HexDump.before.and.after.unblock.zip)

**Comment by steve6375** [comment]
Thanks very much for those dumps. They showed much a simpler structure for the MFT record!
So there were only two attributes which showed a difference (one changed, one deleted). 
The Zone Identifier which was deleted in the fixed version (of course).

To prove that the Zone Identifier is the cause (and not the other field which was changed), please could you repeat the download step you did before (delete the new fixed ISO) as follows:

1. Download a new file under exactly the same conditions as the previous file in the post above (after deleting the previous, now fixed jammy iso)
2. Load up WinHex (Not in read-only mode) and check that the Zone Identifier section is the **last** **section** - there should be FF FF FF FF and loads of 0's after it - if not then download the file again until you can get the Zone field at the end.  It doesn't matter if it is not at the same address as in the example attached, it just must be the last attribute.
3. Check it fails under Ventoy
4. Use WinHEx to change the first four bytes in the Zone Identifier field from 80 00 00 00   to FF FF FF FF - see attached example
5. Save the changes to the disk (this writes to the disk!)
6. See if it fails under Ventoy

I am pretty sure (99.9%) this wont mess up your disk but you can delete the file if it does or repair the NTFS file system under Windows as NTFS if very robust.

If the file then works under Ventoy we have really narrowed down the issue to that attribute and hopefully it can be fixed and a problem that has been around for many years can be resolved at last!





![image](https://user-images.githubusercontent.com/1469350/212721838-fa62fcdb-eb1a-4779-a3b6-96e993eaf81f.png)

**Comment by LaurentGrenet** [comment]
Hi @steve6375 

some good news, but also some bad news.

I begin with bad news : I've not able to update the $MFT as you asked for, because "Write" is impossible with evaluation version of WinHex....

Let me now recap what I did (and give you some good news)

1) Main good news is the fact that, for the 1st time, the iso file I downloaded (with Edge) to do the test you were asking for was DIRECTLY OK (ie. bootable), without needing to copy it, or to remove ASD !
I have then kept a copy of 1024 bytes Hexdump of its $MFT : DirectlyOKHexDump file attached
I have also run the ntfsinfo on it : DirectlyOK.ntfsinfo.txt file attached

2) I then download once more an iso file (with Firefox, this time). This one is (as usual, but now we cannot any more say "as always") NOT bootable.
I have kept a copy  of 1024 bytes Hexdump of its $MFT : NotOKHexDumpBeforeModif
And run ntfsinfo on it : NotOK.ntfsinfo.txt

3) But when I wanted to do the modification of its $MFT, I first detected that the place you asked me to write FF FF FF FF was NOT 80 00 00 00 as you mentioned, but 40 00 00 00
I decided nevertheless to replace this 40 00 00 00 by FF FF FF FF.... but you know already how it eventually ended....

And then of course, I've not been able to test if the file so modified became or not bootable...

Nevertheless, I would be surprised if it was the case, since the "Directly OK" file mentioned above owns this ASD (and is nevertheless bootable)
Moreover, when we copy a non bootable file to a new place, the copy is always bootable, and nevertheless contains the same ASD  than the origin file...
For all these reasons, I guess there is more chance that the key of the problem is rather in the other attribute you mention and that is changed when removing the ASD, than in the removing of ASD itself.... 

Let me remind you that in a post a little bit above (https://github.com/ventoy/Ventoy/issues/2139#issuecomment-1374911124) I already provided you with HexDumps of non bootable files and their bootable copies (that contain the ASD : no "unblock" action on these copies)

I'm afraid you are not at the end of your investigations.....

[Files.zip](https://github.com/ventoy/Ventoy/files/10429264/Files.zip)

**Comment by steve6375** [comment]
Thanks for trying that.
The other file and the copy had many differences.
By editing the file directly it would have proved that the zone identifier record was really the problem. Of course, it is not the existence of that attribute alone, it may be something to do with it's contents or it's position in the record as it is always (?) at the end.
Or it may be the other record that is the problem ...

**Comment by LaurentGrenet** [comment]
Hi @steve6375 

I have news (good and less good) for you.

I found an Hex Editor that is freeware, and allow direct disk update, in particular in MFT records.
I found it here : https://www.disk-editor.org/index.html

I then tried to do your test with this alternative soft.
I replaced the "80 00 00 00" bytes from your screen copy with "FF FF FF FF"
(BTW : When I said yesterday in my post above that I didn't find them, but rather "40 00 00 00", it was an error : I didn't check exactly at the right place...)

I did this unique modification.
As a 1st result, the "Properties" window on this file (from Windows explorer) didn't show any more the file as coming from Internet, as if the ASD were no more there.

Then I tried to boot on the file with Ventoy.
Boot failed, but the error message wasn't exactly the same than before : 
![2023-01-17 14_07_22-Dell-Laurent (Tests boot Ventoy)  En fonction  - Oracle VM VirtualBox](https://user-images.githubusercontent.com/32399301/212912226-2aea5893-89d1-4828-9593-b790112da5ec.png)

But when I went out of the VM, that implied automatic unplug of USB disk from the VM then re-plug on the physical laptop, Windows detected errors on the USB disk. 
I then accepted a "test and repair".

And after this "disk repair".... **the file becomes BOOTABLE in Ventoy !**

I then created a new copy of the HexDump of the file (attached to this post)
But clearly, the "disk repaired" had several impact on this MFT entry, since we can see now several other modifications than the "only" replacement of 80000000 by FFFFFFFF, if we compare it with the Hexdump I did yesterday just after download.

You can find attached the two hexdump of this file
- The one I did yesterday
- and the one I did today, after manual update AND disk repair
Unfortunately, I didn't keep an Hexdump just after my manual update, and before Disk Repair. But I cannot imagine that we would have seen any other difference than the replacement of 80000000 with FFFFFFFF

[Files.zip](https://github.com/ventoy/Ventoy/files/10435621/Files.zip)

**Comment by steve6375** [comment]
What VM are you using? 
VBOX
VBOX+VMUB
VMWARE
QEMU
QEMU via RMPrepUSB
Something else?

**Comment by LaurentGrenet** [comment]
Oracle VirtualBox ver 6.1.40

VMUB ? I don't think so, but what is it ? May be I use it without knowing ?

**Comment by steve6375** [comment]
Well VMUB  causes disk to be fully unlocked by Windows so that VBox will mount is as rd/wr physical drive and not as a snapshotted drive which can cause issues.

Anyway, maybe you can repeat a download, make a bad file, check if fails and Zone Ident is at the **end** of the record, then modify it. Then unplug and replug it and re-check it (Properties - Tools - check). Then check record is still edited OK.
Then make **new** **VMDK** for the drive and test with Ventoy under VBox - OR don't use VBOX bit always use a real system?

or use VBOX v5 + VMUB which is great for testing booting from USB drives!
https://rmprepusb.com/tutorials/004-how-to-boot-directly-from-a-usb-drive-using-an-emulator-or-vm-under-windows/#Enabling_full_rdwr_access_with_VMUB_highly_recommended

**Comment by LaurentGrenet** [comment]
As per your demand, I have done a new test.

1) Download (with Firefox) a new file
2) Yes, it is "bad"  
3) Get a copy of MFT record : Hexdump initial.txt
Is ZoneIdentifier at the end ? It seems to me that yes, but ZoneIdentifier is mentioned twice in the HexDump, as you can see. So I don't know and let you appreciate this point
4) Modify the record, and keep a copy after modif : HexDump after modif before unplug.txt
5) Exit the VM used to get MFT record (with DiskEditor, and no more with WinHex, which explain that format of copies are not the same than in previous tests)
6) Unplug the disk
7) Plug the disk
8) Properties/Tools/Check => repair needed
9) Repair done
10) Get a copy of the MFT record : Hexdump after unplug-plug-check-repair.txt
You can see that this new Hexdump after repair is **very different** from the previous one 
11) Try to boot the **REAL** laptop with Ventoy on this disk
- browsing the disk under ventoy to select the ISO, it appears that file has length 0 (what I didn't catch before... because I didn't check it. Nevertheless, I'm quite confident it has been set to length 0 by the repair...)
- I nevertheless try to boot on it, but (no surprising) this leads to the error
```
error: image chunk is null
.
VVVVVVVV 2 grub_errno=0
ventoy not ready
chain empty failed
press Enter to continue....

```
12) Reboot the REAL laptop on W10
13) Yes, I confirm, seen under W10, file is length 0
14) Get a last copy of MFT record : Hexdump Last copy.txt
Identical with copy just after repair

I let you to dig in these files, but for me, I begin to be a little bit tired with these tests, and I'll stop here.
And this even more since we know that this problem in ventoy is NOT a blocking one : We have several ways to make a "non bootable" file "bootable". So, if I can understand it is important FOR YOU to know WHY some files are not bootable, I'm obliged to recognize that personnally... I don't really care....

Regards, and good luck to find the root cause !
[Files.zip](https://github.com/ventoy/Ventoy/files/10438214/Files.zip)

**Comment by steve6375** [comment]
Ok thanks, I totally understand and I thank you very much for your tremendous efforts.
Best wishes
Steve
